### DISCUSSION TITLE
NVIDIA Shield Bluetooth Headphone Connect Script

### DISCUSSION THREAD

**Discussion by michael7oliver** [PRProd/HA-Firemote#160] [Q&A] [answered]
I feel like I might be pushing my luck asking for more help but another script I thought would be great to implement is connecting the NVIDIA Shield to an already paired set of Bluetooth headphones. I only use one set with my shield but as I also use them with my phone they typically connect to the phone not the shield when I turn them on. A script would save digging through the setting every time I use them. 

I’ve had a play around with this but I can’t figure out how to define this action in HA.

**Comment by PRProd** [comment]
@michael7oliver - Pushing your luck?  No.  Not at all!  :smile:  Questions like yours are helpful for other people, and they sometimes lead into useful enhancements for the Firemote project!  

I'll dig into this and see what I can come up with... just as soon as I can locate where my bluetooth headphones are hiding

**Comment by PRProd** [comment]
My headphones happily connected somehow to both my Shield and my phone.  No matter what I did, I couldn't reproduce what you described.

I did try digging around for a while and I might have something useful.  Give this command a try
```
adb shell am start -n com.android.tv.settings/.accessories.AddAccessoryActivity
```

You can try the command in this menu, just like in our previous conversations
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/c423958a-65a3-4071-9f8e-a9c93a07aad1)


While I'm not completely sure that this is what you were looking for, maybe it's a start?

**Reply by michael7oliver** [reply]
Thanks - it's really close. This command starts the bluetooth pairing screen. My headphone are already paired but for whatever reason they preferentially connect to my phone and not the NVIDIA shield. 

I'm trying to work out where you're getting these commands from so I can see if I can solve it myself but I'm not having any luck. 

I think you might be able to recreate my situation by selecting your headphones from the "settings / remotes & accessories" menu and choosing to disconnect the device. Then you'll need to choose to connect to them as I do... (note I'm not in the habit of disconnecting mine... just looking for a way to recreate the situation for you).

**Comment by PRProd** [comment, marked as answer]
@michael7oliver 

I tried really really hard here again, and I'm not extremely happy with the solution I've come up with this time, but I think it might do the trick.  Try this:
```
adb shell am start -n com.android.tv.settings/.MainSettings && (input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20) && input keyevent 66 && (input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20)
```

This command should bring you to the exact place where you would select your paired headphones, but it doesn't go any further.  If this brings you to the right place, we can also add a couple additional commands to complete the connection process.

**Reply by michael7oliver** [reply]
Thanks Doug, I’ll try it out this evening.

**Reply by PRProd** [reply]
Any luck?

**Reply by michael7oliver** [reply]
Hey thanks for the reminder. I was busy watching the cricket and playing Zelda last night and forgot to try it...

Just tried it but for my case I had to make a few changes... one less down stroke at the start and I added a button press at the end to connect them i.e. input key event 66. I can see why you weren't happy with it but it works perfectly, really happy with it. Thanks again :). Just as importantly I've learned a bit too!

`adb shell am start -n com.android.tv.settings/.MainSettings && (input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20) && input keyevent 66 && (input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20 & input keyevent 20) && input keyevent 66 && (input keyevent 20 & input keyevent 20) && input keyevent 66`

**Reply by PRProd** [reply]
Learning is great!  Sharing what you have learned is even better!

Thanks for letting me help!
